midterm-2023-11-07/code/5311-kristijan-lazarov/script.ts

```diff
@@ -0,0 +1,322 @@
+interface Movie {
+    id: number
+    title: string
+    year: number
+    director: string
+    genre: string[]
+    plot: string
+    cast: Actor[]
+    oscars: any
+    rating: number
+}
+
+interface Actor {
+    actor: string
+    character: string
+}
+
+
+type MovieSorter = (first: Movie, second: Movie) => number;
+document.addEventListener('DOMContentLoaded', siteCode);
+let movies: Movie[] = [];
+
+const idSorter: MovieSorter = (first, second) => first.id - second.id;
+const nameSorter: MovieSorter = (first, second) => first.title.localeCompare(second.title);
+const directorSorter: MovieSorter = (first, second) => first.director.localeCompare(second.director);
+const yearSorter: MovieSorter = (first, second) => first.year - second.year;
+const oscarSorter: MovieSorter = (first, second) => {
+    const firstOscars = Object.keys(first.oscars).length;
+    const secondOscars = Object.keys(second.oscars).length;
+
+    return secondOscars - firstOscars;
+}
+
+async function siteCode() {
+    const data = await loadData();
+    movies = data;
+
+    fillterGenres(movies);
+
+    displayMovies(movies);
+
+
+    const idSort = document.getElementById('sort-id') as HTMLElement;
```

Comment:
We could have an additional class, for example and use the "data-id" parameter from the HTML to avoid writing all this repetitive code.